src/components/text-input/TextInput.scss

```diff
@@ -1,55 +1,107 @@
 @import '../../styles/variables';
+@import '../../styles/inputs';
 
-/**************************************
- * Text Input
- **************************************/
+/** @define TextInput */
+
+.bdl-TextInput-input {
+    @include common-typography;
+
+    height: 32px;
+    margin-top: 5px;
+
+    &[type="text"],
+    &[type="password"],
+    &[type="search"],
+    &[type="email"],
+    &[type="url"],
+    &[type="tel"],
+    &[type="number"] {
+        @include box-inputs;
+
+        width: 262px;
+
+        &:hover {
+            @include box-inputs-hover;
+        }
+
+        &:focus {
+            @include box-inputs-focus;
+        }
+
+        &:disabled {
+            background-color: $haze;
+            border-color: $efive;
+            box-shadow: none;
+            color: $bfive;
+
+            &:hover {
+                border-color: $efive;
+            }
+        }
+    }
+
+    // IE11 adds a new x to the inputs to clear it
+    &::-ms-clear {
+        display: none;
+    }
+
+    &::-webkit-input-placeholder,
+    &::-moz-placeholder,
+    &::-ms-input-placeholder {
+        color: $seesee;
+    }
+
+    /* Safari does't treat search input like a normal textfield */
+    &[type="search"] {
+        appearance: textfield;
+
+        &::-webkit-search-cancel-button,
+        &::-webkit-search-results-button {
+            display: none;
+        }
+    }
+
+    &[type="number"] {
+        appearance: textfield;
+
+        &::-webkit-outer-spin-button,
+        &::-webkit-inner-spin-button {
+            -webkit-appearance: none;
+            margin: 0;
+        }
+    }
+}
+
+.bdl-TextInput-container {
+    @include common-typography;
 
-.text-input-container {
     margin: 0 0 20px;
     position: relative;
 
-    .text-input-loading {
+    &.bdl-has-error {
```

Comment:
Should this be `.bdl-hasError`?

Reply:
Yes!

Reply:
Actually is/has is followed by a - and thereafter its camelCase.

With namespace:
.bdl-is-loading, .bdl-has-error, .bdl-is-errorShown, .bdl-has-criticalError.

Without namespace:
.is-loading, .has-error, .is-errorShown, .has-criticalError.

We can however maybe enforce just `is` and drop `has` so in the above examples

.has-criticalError --> .is-criticalErrorShown